util/util.go

```diff
@@ -51,6 +52,8 @@ var (
 	rnd                          = rand.New(rand.NewSource(time.Now().UnixNano()))
 	ErrNoCluster                 = fmt.Errorf("no %q label present", clusterv1.ClusterLabelName)
 	ErrUnstructuredFieldNotFound = fmt.Errorf("field not found")
+	// taken from k8s.io/cmd/kubeadm/app/util
+	kubeReleaseRegex = regexp.MustCompile(`^v?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)([-0-9a-zA-Z_\.+]*)?$`)
```

Comment:
Can we use this regex in OpenAPI validation on the k8s version field so we can ensure proper inputs?

Reply:
Wouldnt this be a duplicated of the code implemented in the validation webhook?